content/guides/getstarted/intro-to-ai.md

```diff
@@ -0,0 +1,544 @@
+4. Different clients like X, telegram, discord. etc.
+
+Its takes all these things and puts them together in a way that you can create a
+"Character" that can interact with users or autonomously. These Characters are
+defined in a JSON file, which defines which actions it can perform, which LLM it
+should use and different prompts that define the behaviour of the character.
+
+### Building a Twitter (X) bot that can mint NFTs on request
+
+What we will build now is a helpful solana developer assistant that can help
+developers on the platform X to get started with solana and it will also be able
+to mint NFTs to the users on demand.
+
+<Steps>
+
+### Clone the repository
+
+First clone the Eliza repository:
+
+```bash
+git clone https://github.com/elizaOS/eliza.git
+```
+
+### Set up the environment
+
+Make sure you have `node` version `23.x.x` installed using
+[nvm](https://github.com/nvm-sh/nvm):
+
+```bash
+node --version
+nvm use 23
+```
+
+Then install the dependencies:
+
+```bash
+pnpm install --no-frozen-lockfile
+pnpm build
+```
+
+### Configure environment variables
+
+Create your environment file:
+
+```bash
+cp .env.example .env
+```
+
+What we are interested in for this guide is:
+
+```env
+TWITTER_USERNAME=               # Account username
+TWITTER_PASSWORD=               # Account password
+TWITTER_EMAIL=
+OPENAI_API_KEY=                 # Get this from: https://platform.openai.com/api-keys
+SOLANA_RPC_URL=https://api.devnet.solana.com # Lets work on devnet for now
+SOLANA_PRIVATE_KEY=             # Get this from: https://solana.com/wallet
+SOLANA_PUBLIC_KEY=
+SOLANA_CLUSTER=devnet # Default: devnet. Solana Cluster: 'devnet' | 'testnet' | 'mainnet-beta'
+SOLANA_ADMIN_PRIVATE_KEY=       # This wallet is used to verify NFTs (you can use the same as the SOLANA_PRIVATE_KEY for now)
+SOLANA_ADMIN_PUBLIC_KEY=        # This wallet is used to verify NFTs (you can use the same as the SOLANA_PUBLIC_KEY for now)
+```
+
+Make sure you have some
+[devnet sol](https://solana.com/de/developers/guides/getstarted/solana-token-airdrop-and-faucets)
+in your wallet before you start the agent.
+
+### Encode private key (if needed)
+
+If you have your private key in hex format you can base58 encode your private
+key using the following js snippet:
+
+```ts filename=encode58.ts
+const bs58 = require('bs58');
+console.log(bs58.encode(new Uint8Array([213,29,143,...])));
+```
+
+And run it using
+
+```bash
+npx tsx encode58.ts
+```
+
+### Create character configuration
+
+To create a new character I would recommend using the composer feature in Cursor
+and tell it what kind of character you want to create. Give it one of the other
+characters as input and describe what kind of character you want to create.
+
+```json filename=character.json
+{
+    "name": "Sol",
+    "clients": ["twitter"],
+    "modelProvider": "openai",
+    "settings": {
+        "voice": {
+            "model": "en_US-neural"
+        },
+        ...
+```
```

Comment:
Where is this file coming from, where is it supposed to go, and what does the full json format look like? Maybe add a link to a reference?

Reply:
Added more explanation and the file path to find these files.